crates/rbuilder/src/live_builder/order_input/mod.rs

```diff
@@ -215,12 +225,18 @@ where
         global_cancel.clone(),
     )
     .await?;
-    let txpool_fetcher = txpool_fetcher::subscribe_to_txpool_with_blobs(
-        config.clone(),
-        order_sender.clone(),
-        global_cancel.clone(),
-    )
-    .await?;
+    let txpool_fetcher = if config.ipc_path.is_some() {
+        info!("IPC path configured, starting txpool subscription");
+        txpool_fetcher::subscribe_to_txpool_with_blobs(
+            config.clone(),
+            order_sender.clone(),
+            global_cancel.clone(),
+        )
+        .await?
+    } else {
+        info!("No IPC path configured, skipping txpool subscription");
+        tokio::spawn(async {}) // Empty task
```

Comment:
Instead of this, can you have an array of async handles and append this one if IPC is enabled?
```
let handles = [clean_job, rpc_server]

if ipc_is_enabled {
   let txpool_fetcher = subscribe_to_txpool...
   handles.add(txpool_fetcher)
}
```

Reply:
Yes, can you please check the new commit